src/main/java/cubicchunks/asm/mixin/core/common/MixinWorld.java

```diff
@@ -95,6 +97,10 @@
 		this.minHeight = CubicChunks.Config.DEFAULT_MIN_WORLD_HEIGHT;
 		//don't want to make world implement IConfigChangeListener
 		CubicChunks.addConfigChangeListener(this);
+
+		if(!(this.provider instanceof ICubicWorldProvider)) { // if the provider is vanilla, wrap it
+			this.provider = new VanillaCubicProvider(this, provider, null);
```

Comment:
The reason I added that extra argument to VanillaCubicProvider was to recycle the old chunk generator. (kind of a pre optimization... the garbage collector would get it anyway so its not a wast of memory)

Anyway, thanks for reminding me, I will clean up VanillaCubicProvider. :)